test-fips-docker.sh

```diff
@@ -0,0 +1,463 @@
+        missing_deps+=("wget")
+    fi
+    
+    # Check for mvn (Maven)
+    if ! command -v mvn > /dev/null 2>&1; then
+        missing_deps+=("maven")
+    fi
+    
+    # Check for curl
+    if ! command -v curl > /dev/null 2>&1; then
+        missing_deps+=("curl")
+    fi
+    
+    if [ ${#missing_deps[@]} -ne 0 ]; then
+        print_status "ERROR" "Missing required dependencies: ${missing_deps[*]}"
+        echo "Please install the missing dependencies and try again."
+        exit 1
+    fi
+}
+
+# Check if Docker is available
+if ! command -v docker > /dev/null 2>&1; then
+    print_status "ERROR" "Docker is not installed. Please install Docker and try again."
+    exit 1
+fi
+
+print_status "SUCCESS" "Docker is available ($(docker --version))"
+
+# Check prerequisites
+check_prerequisites
+
+# Cleanup existing containers and images if requested
+cleanup_existing
+
+# Clean up any existing container with the same name
+docker stop geoserver-fips 2>/dev/null || true
+docker rm geoserver-fips 2>/dev/null || true
+
+# Function to download Jetty distribution
+download_jetty() {
+    if [ "$SKIP_DOWNLOAD" = true ]; then
+        print_status "WARNING" "Skipping Jetty download (--skip-download specified)"
+        return 0
+    fi
+    
+    local jetty_file="jetty-distribution-9.4.44.v20210927.tar.gz"
+    local jetty_url="https://repo1.maven.org/maven2/org/eclipse/jetty/jetty-distribution/9.4.44.v20210927/jetty-distribution-9.4.44.v20210927.tar.gz"
+    
+    if [ ! -f "$jetty_file" ]; then
+        echo "Downloading Jetty distribution..."
+        local retry_count=0
+        local max_retries=3
+        
+        while [ $retry_count -lt $max_retries ]; do
+            if wget -q --timeout=30 "$jetty_url" -O "$jetty_file"; then
+                print_status "SUCCESS" "Jetty distribution downloaded successfully"
+                return 0
+            else
+                retry_count=$((retry_count + 1))
+                if [ $retry_count -lt $max_retries ]; then
+                    print_status "WARNING" "Download failed, retrying... (attempt $retry_count/$max_retries)"
+                    sleep 2
+                else
+                    print_status "ERROR" "Failed to download Jetty distribution after $max_retries attempts"
+                    return 1
+                fi
+            fi
+        done
+    else
+        print_status "SUCCESS" "Jetty distribution already exists"
+    fi
+}
+
+# Function to build GeoServer WAR file
+build_geoserver() {
+    if [ "$SKIP_BUILD" = true ]; then
+        print_status "WARNING" "Skipping GeoServer build (--skip-build specified)"
+        return 0
+    fi
+    
+    local war_file="src/web/app/target/geoserver.war"
+    
+    if [ ! -f "$war_file" ]; then
+        echo "Building GeoServer WAR file..."
+        if [ -d "src" ]; then
+            cd src
+            echo "Running Maven build (this may take several minutes)..."
+            if mvn clean package -DskipTests -q; then
```

Comment:
we should remove skipTests so all the unit tests run with this. Or at least make sure to run them in some other step of the build process.